components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -79,53 +79,79 @@ public class LegacyJobOrchestrator : IJobOrchestrator {
         /// <param name="ct"></param>
         /// <returns></returns>
         public Task<JobProcessingInstructionModel> GetAvailableJobAsync(string workerId, JobRequestModel request, CancellationToken ct = default) {
-            if (_assignedJobs.TryGetValue(workerId, out var job)) {
+            try {
+                _lock.Wait(ct);
+                if (_assignedJobs.TryGetValue(workerId, out var job)) {
+                    return Task.FromResult(job);
+                }
+                if (_availableJobs.Count > 0 && _availableJobs.TryDequeue(out job)) {
+                    _assignedJobs.AddOrUpdate(workerId, job);
+                }
+
                 return Task.FromResult(job);
             }
-            if (_availableJobs.Count > 0 && _availableJobs.TryDequeue(out job)) {
-                _assignedJobs.AddOrUpdate(workerId, job);
-                if (_availableJobs.Count == 0) {
-                    _updated = false;
-                }
+            catch(OperationCanceledException) {
+                JobProcessingInstructionModel job = null;
+                return Task.FromResult(job);
```

Comment:
Why do we return `null` when there was a cancellation and don't propagate the error up ?

Reply:
propagated OperationCanceledExceptions to caller for both SendHeartbeatAsync and GetAvailableJobAsync